### PR TITLE
Add check for result codes from HMI IsReady response

### PR DESCRIPTION
**Pull request by ychernysheva**
Fixes #[1384](https://github.com/smartdevicelink/sdl_core/issues/1384) and #3205

This PR is **not ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
This pull request is intended to fix issue with incorrect determination of interface availability, when HMI sends IsReady response. Corresponding to requirements, message with any error result code and "available" = "true" is equal to "available" = "false", when interface is set as unavailable. That's why check of result codes from IsReady response is added to ChangeInterfaceState method, also minor changes are added to corresponding requests for processing RPC to align their responses to requirements.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/src/commands/request_to_hmi.cc [hunk 1]
```diff
@@ -50,13 +51,40 @@ bool ChangeInterfaceState(ApplicationManager& application_manager,
...
+              response_code,
+              hmi_apis::Common_Result::SUCCESS,
+              hmi_apis::Common_Result::WARNINGS)) {
+        application_manager.hmi_interfaces().SetInterfaceState(
+            interface, HmiInterfaces::STATE_NOT_AVAILABLE);
+        return false;
+      }
+      application_manager.hmi_interfaces().SetInterfaceState(
+          interface, HmiInterfaces::STATE_AVAILABLE);
+      return true;
```
Comment (IGapchuk) [inline src/components/application_manager/src/commands/request_to_hmi.cc:80]: @ychernysheva I believe the part from 78 until 80 lines is redundant. In the code below we set the same parameters and return the `is_available` variable that has `true` value at this moment.

File: src/components/application_manager/src/commands/request_to_hmi.cc [hunk 2]
```diff
@@ -50,13 +51,40 @@ bool ChangeInterfaceState(ApplicationManager& application_manager,
...
+    }
+
+    application_manager.hmi_interfaces().SetInterfaceState(
+        interface, HmiInterfaces::STATE_AVAILABLE);
     return is_available;
```
Comment (IGapchuk) [inline src/components/application_manager/src/commands/request_to_hmi.cc:85]: @ychernysheva I think the lines 83-85 should should replace the lines 78-80.

File: src/components/application_manager/src/commands/request_to_hmi.cc [hunk 3]
```diff
@@ -50,13 +51,40 @@ bool ChangeInterfaceState(ApplicationManager& application_manager,
...
+              response_from_hmi[strings::params][hmi_response::code].asInt());
+
+      using helpers::Compare;
+      using helpers::EQ;
+      using helpers::ONE;
+
+      if (!Compare<hmi_apis::Common_Result::eType, EQ, ONE>(
+              response_code,
+              hmi_apis::Common_Result::SUCCESS,
+              hmi_apis::Common_Result::WARNINGS)) {
```
[the remark this answers is not in the document]
Comment (ychernysheva) [reply src/components/application_manager/src/commands/request_to_hmi.cc]: Done in dce64912ffc289f48ad2dcce0ec341334b0715c6

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/commands/mobile/subscribe_vehicle_data_request.cc [hunk 4]
```diff
@@ -287,8 +287,8 @@ void SubscribeVehicleDataRequest::CheckVISubscriptions(
     smart_objects::SmartObject& out_response_params,
     smart_objects::SmartObject& out_request_params,
     bool& out_result) {
-  const bool is_interface_not_available =
-      (HmiInterfaces::STATE_NOT_AVAILABLE ==
+  const bool interface_available =
```
[the remark this answers is not in the document]
Comment (ychernysheva) [reply src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/commands/mobile/subscribe_vehicle_data_request.cc]: Done in dce64912ffc289f48ad2dcce0ec341334b0715c6

File: src/components/application_manager/src/commands/request_to_hmi.cc [hunk 5]
```diff
@@ -50,13 +51,40 @@ bool ChangeInterfaceState(ApplicationManager& application_manager,
...
+    }
+
+    if (response_from_hmi[strings::params].keyExists(hmi_response::code)) {
+      hmi_apis::Common_Result::eType response_code =
+          static_cast<hmi_apis::Common_Result::eType>(
+              response_from_hmi[strings::params][hmi_response::code].asInt());
+
+      using helpers::Compare;
+      using helpers::EQ;
+      using helpers::ONE;
```
Comment (mkorniichuk) [inline src/components/application_manager/src/commands/request_to_hmi.cc:68]: @ychernysheva I would move this to the very beginning of the function

**Pull request closed without merging**